saq/queue/postgres.py

```diff
@@ -0,0 +1,521 @@
+                WHERE status = 'active'
+                  AND objid IS NULL
+                """
+            )
+            results = await cursor.fetchall()
+        swept = []
+        for result in results:
+            job = self.deserialize(result[0])
+            if not job:
+                continue
+            swept.append(job.key)
+            if job.retryable:
+                await self.retry(job, error="swept")
+            else:
+                await job.finish(Status.ABORTED, error="swept")
+        return swept
+
+    async def listen(
+        self,
+        job_keys: Iterable[str],
+        callback: ListenCallback,
+        timeout: float | None = 10,
+    ) -> None:
+        if not job_keys:
+            return
+
+        async with self.pool.connection() as conn:
+            for key in job_keys:
+                await conn.execute(f'LISTEN "{key}"')
+            await conn.commit()
+            gen = conn.notifies(timeout=timeout or None)
+            async for notify in gen:
+                payload = self._load(notify.payload)
+                key = payload["key"]
+                status = Status[payload["status"].upper()]
+                if asyncio.iscoroutinefunction(callback):
+                    stop = await callback(key, status)
+                else:
+                    stop = callback(key, status)
+
+                if stop:
+                    await gen.aclose()
+
+    async def notify(self, job: Job) -> None:
+        payload = self._dump({"key": job.key, "status": job.status})
+        await self._notify(job.key, payload)
+
+    async def _notify(self, channel: str, payload: t.Any) -> None:
+        async with self.pool.connection() as conn:
+            await conn.execute(f"NOTIFY \"{channel}\", '{payload}'")
+
+    async def update(self, job: Job) -> None:
+        job.touched = now()
+        async with self._get_connection(job.key) as conn:
+            await conn.execute(
+                f"""
+                    UPDATE {self.jobs_table} SET job = %(job)s, status = %(status)s
+                    WHERE key = %(key)s
+                    """,
+                {"job": self.serialize(job), "status": job.status, "key": job.key},
+            )
+        await self.notify(job)
+
+    async def job(self, job_key: str) -> Job | None:
+        async with self._get_connection(job_key) as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                f"""
+                SELECT job FROM {self.jobs_table}
+                WHERE key = %(key)s
+                """,
+                {"key": job_key},
+            )
+            job = await cursor.fetchone()
+            if job:
+                return self.deserialize(job[0])
+        return None
+
+    async def abort(self, job: Job, error: str, ttl: float = 5) -> None:
+        job.error = error
+        job.status = Status.ABORTING
+        job.ttl = int(seconds(now()) + ttl)
```

Comment:
why are you setting the job ttl here? i don't think redis does this. redis usses the abort ttl for the abort key right?

Reply:
Redis also sets the job expiration to abort ttl + 1. I can update this to match what Redis is doing.